lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -305,7 +305,7 @@ def merge_credit_card_options(parameters, options)
         parameters[:credit_card] ||= {}
         parameters[:credit_card].merge!(:options => valid_options)
         address = options[:billing_address]&.except(:phone)
-        return parameters if address.nil? || address.empty?
+        return parameters if address.nil? || address.values.compact.empty?
```

Comment:
`address.all? { |_k, v| v.nil? || v.empty? }` will avoid creating the two extra `Array`s you're making with the `.values.compact` sequence. I'm not sure we care in this particular case, though.

Reply:
Noted. I wonder if the processing time saved would actually be more than it takes to run remote tests again!

Reply:
Probably not. This kind of thing adds up, especially because Ruby's memory-hungry and doesn't have the best GC ever, but when you've got super-small arrays that are very short-lived (which is the exact case here), the benefit is exceedingly minimal.